### PR TITLE
nixos/qtile: fixes

### LINKED ISSUE #167328: Cannot use globally defined Python environment with Qtile
### Describe the bug
When enabling the Qtile window manager, globally defined Python environments cannot be used.

### Steps To Reproduce
1. Declaratively  define a global Python environment in `configuration.nix`:
    
    ```nix
    environment.systemPackages = with pkgs; [
       (let
          my-python-packages = python-packages: with python-packages; [
            matplotlib
            numpy
            pandas
            scikit-learn
          ];
          python-with-my-packages = python3.withPackages my-python-packages;
        in
          python-with-my-packages)
    ];
    ```
2. Test Python environment:
   
    ```console
    $ python -c "import numpy"
    $ python -c "import sys; print(sys.prefix)"
    /nix/store/kc9bsiwsbadhs165ryr4b6xbsb0wccmc-python3-3.9.6-env
    ```
3. Enable Qtile:

    ```nix
    services.xserver.windowManager = {
      qtile = {
        enable = true;
      };
    };
    ```
4. Previously defined Python environment is no longer available:
    
   ```console
   $ python -c "import numpy"
   Traceback (most recent call last):
    File "<string>", line 1, in <module>
   ModuleNotFoundError: No module named 'numpy'
   $ python -c "import sys; print(sys.prefix)"
   /nix/store/z43s37a755b2i8nd6mibjzjjjjdl7f3n-qtile-0.18.1
   ```

### Expected behavior
I expect to be able to use globally defined Python environments regardless of the window manager in use, in this case, Qtile. 

### Additional context
This issue has been briefly discussed in NixOS [discourse](https://discourse.nixos.org/t/using-qtile-changes-python-environment/18472).

### Notify maintainers
@kamilchm @jonringer 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.32, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.16`
 - channels(loren): `"nixos-21.11-21.11.335481.604c44137d9"`
 - channels(root): `"nixos-21.11.336820.af0a9bc0e53, nixos-unstable-22.05pre365028.710fed5a248"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by jonringer**
###### Description of changes

closes: #167328

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/x11/window-managers/qtile.nix [hunk 1]
```diff
@@ -7,19 +7,34 @@ let
...
+    enable = mkEnableOption "qtile";
+
+    package = mkOption {
+      type = package;
+      default = pkgs.qtile;
+      defaultText = "pkgs.qtile";
+      description = ''
+          The qtile package to be ran.
+      '';
+    };
```
Comment (pennae) [inline nixos/modules/services/x11/window-managers/qtile.nix:20]:
```suggestion
    package = mkPackageOption pkgs "qtile" {};
```

File: nixos/modules/services/x11/window-managers/qtile.nix [hunk 2]
```diff
@@ -7,19 +7,34 @@ let
...
         waitPID=$!
       '';
     }];
 
-    environment.systemPackages = [ pkgs.qtile ];
+    environment.systemPackages = [
+      # pkgs.qtile is currently a buildenv of qtile and its dependencies.
+      # For userland commands, we want the underlying package so that
+      # packages such as python don't bleed into userland and overwrite intended behavior.
+      (pkgs.qtile.unwrapped or pkgs.qtile)
```
Comment (pennae) [inline nixos/modules/services/x11/window-managers/qtile.nix:37]: shouldn't this use `cfg.package.unwrapped` as its first choice? (no idea, not using qtile)

Comment (jonringer) [reply nixos/modules/services/x11/window-managers/qtile.nix:37]: yea, it should

File: nixos/modules/services/x11/window-managers/qtile.nix (hunk 1 above)
Comment (jonringer) [reply nixos/modules/services/x11/window-managers/qtile.nix:20]: didn't know this existed. Thanks

**Comment by nieomylnieja** [comment]
are you sure this actually fixes the issue? It does not seem to work: https://github.com/NixOS/nixpkgs/issues/186243

**Pull request merged**